In [9]:
import numpy as np
import scipy.stats as stats


1. Даны две  независимые выборки. Не соблюдается условие нормальности<br>
x1  380,420, 290\
y1 140,360,200,900\
Сделайте вывод по результатам, полученным с помощью функции

In [10]:
x1 = np.array([380,420,290])
y1 = np.array([140,360,200,900])
alpha = 0.05
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue=0.6285714285714286 > alpha, применяем нулевую гипотезу. <br> Статистически значимых различий нет. 

2. Исследовалось влияние препарата на уровень давления пациентов. <br>
Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. <br>
Есть ли статистически значимые различия? <br>
1е измерение до приема препарата: 150, 160, 165, 145, 155 <br>
2е измерение через 10 минут: 140, 155, 150,  130, 135 <br>
3е измерение через 30 минут: 130, 130, 120, 130, 125 

In [11]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])
alpha = 0.05
stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.00831683351100441 < alpha, применяем альтернативную гипотезу. <br> Статистически значимые различия есть. <br>
Исследование показало, что препарат влияет на уровень давления пациентов.

3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [12]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
alpha = 0.05
stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625 > alpha, применяем нулевую гипотезу. <br> Статистически значимых различий нет. <br>
Исследование показало, что препарат не влияет на уровень давления пациентов.

4. Даны 3 группы  учеников плавания. <br>
В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67 <br>
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53 <br>
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54 <br>

In [13]:
x1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
alpha = 0.05
stats.kruskal(x1, x2, x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

pvalue=0.0650380998590494 > alpha, применяем нулевую гипотезу. <br> Статистически значимых различий нет. <br>
Время на дистанцию 50 м одинаковое.

5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. <br>
Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. <br>
Объем выборки 10, уровень статистической значимости 5% <br>
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [14]:
weights = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = 10
mean = weights.mean()  #среднее арифметическое
std = weights.std(ddof=1)
mean, std

(2.5279999999999996, 0.1572542173961923)

Поскольку мы проверяем гипотезу относительно математического ожидания нормально распределённой <br>
случайной величины с неизвестной дисперсией, будем использовать t-критерий.

In [15]:
t = (mean-2.5)/(std/np.sqrt(n))
t

0.5630613661802959

Доверительная вероятность 0.05, число степеней свободы: n - 1 = 9 <br>
по таблице Стьюдента находим критическое: 

In [16]:
stats.t.ppf(0.975, 9)    #9 степеней свободы

2.2621571627409915

Расчетное значение = 0.5630613661802959 > t-наблюдаемого  = 2.2621571627409915* <br>
Принимаем нулевую гипотезу, на уровне статистической значимости 5% статистически значимых различий нет.* 